plugins/org.komodo.shell/src/org/komodo/shell/commands/core/ImportCommand.java

```diff
@@ -60,101 +61,195 @@ public ImportCommand( final WorkspaceStatus wsStatus ) {
-//            	String modelName = theModel.getName(wsStatus.getTransaction());
-//            	modelObject.rename(wsStatus.getTransaction(), modelName);
-//                // Commit transaction
-//                getWorkspaceStatus().commit("ImportCommand"); //$NON-NLS-1$
-//
-//                print(CompletionConstants.MESSAGE_INDENT, Messages.getString("ImportCommand.ModelImportSuccessMsg", modelPathArg, fileNameArg)); //$NON-NLS-1$
-//                if (getWorkspaceStatus().getRecordingStatus())
-//                    recordCommand(getArguments());
-//            } else {
-//                print(CompletionConstants.MESSAGE_INDENT, Messages.getString("ImportCommand.ImportFailedMsg", fileNameArg)); //$NON-NLS-1$
-//                print(CompletionConstants.MESSAGE_INDENT, importMessages.errorMessagesToString());
-//            }
-        } else if (SUBCMD_VDB.equalsIgnoreCase(subcmdArg)) {
-            // Check required args
-            String fileNameArg = requiredArgument(1, Messages.getString("ImportCommand.InvalidArgMsg_FileName")); //$NON-NLS-1$
-            String parentContextPath = optionalArgument(2);
-
-            // Validate fileName
-            if(!validateFileName(fileNameArg)) {
-            	return false;
+        	// Can only import DDL into Model and Schema
+        	KomodoType kType = parentContext.getKomodoObj().getTypeIdentifier(trans);
+        	if(kType != KomodoType.MODEL && kType != KomodoType.SCHEMA) {
+        		print(CompletionConstants.MESSAGE_INDENT, Messages.getString(Messages.ImportCommand.InvalidDDLParentType));
+        		return false;
+        	}
+        	
+            // Import the DDL into the target context
+            File ddlFile = new File(fileNameArg);
+            ImportOptions importOptions = new ImportOptions();
+            ImportMessages importMessages = new ImportMessages();
+
+            WorkspaceContext tempContext = createTempContext(kType,TEMP_IMPORT_CONTEXT);
+            if(tempContext==null) {
+        		print(CompletionConstants.MESSAGE_INDENT, Messages.getString(Messages.ImportCommand.ErrorCreatingTempNode,TEMP_IMPORT_CONTEXT));
+        		return false;
+            }
+            
+            // Setup the import
+            importDdl(getWorkspaceStatus().getTransaction(), ddlFile, tempContext, importOptions, importMessages);
+            
+            if(!importMessages.hasError()) {
+            	
+        		print(CompletionConstants.MESSAGE_INDENT, Messages.getString(Messages.ImportCommand.DdlImportInProgressMsg, ddlFile));
+        		
+            	// The commit will initiate sequencing
+            	wsStatus.commitImport(ImportCommand.class.getSimpleName(), importMessages);
+            	
+            	// No sequencing problems - success
+            	if(!importMessages.hasError()) {
+        			// Move the children underneath the supplied parent context
+                    List<WorkspaceContext> children = tempContext.getChildren();
```

Comment:
Cannot understand the benefit of the temporary context. Why can the import not just import straight under the given parent. Are we certain the sequencers are not running twice, once on the temp context and the other on the final moved-to parent?